extensions/vertx-http/deployment-spi/src/main/java/io/quarkus/vertx/http/deployment/spi/AdditionalStaticResourceBuildItem.java

```diff
@@ -3,11 +3,16 @@
 import io.quarkus.builder.item.MultiBuildItem;
 
 /**
- * Meant to be used by extensions that generate resource into {@code META-INF/resources}.
+ * Meant to be used by extensions that generate resource into {@code META-INF/resources} on
+ * {@link io.quarkus.runtime.LaunchMode#NORMAL} mode, if you want to generate resource into {@code META-INF/resources}
+ * use {@link GeneratedStaticResourceBuildItem} instead.
```

Comment:
I'm finding this a bit confusing. I read this sentence a few times and I don't understand exactly when I should use one and not the other?

We need to be sure people will use the right one so we need things to be extremely clear. BTW, maybe @cescoffier can chime in to improve that as IIUC the general idea came from him.

Reply:
Yeah it's not clear, `AdditionalStaticResourceBuildItem` should not be used by external extensions anymore, we keep it here for backward compat. We could duplicate it in deployment and mark this one as @Deprecated, but that would be too complex no?